runtime/compiler/env/j9method.cpp

```diff
@@ -5839,6 +5840,54 @@ TR_ResolvedJ9Method::longConstant(I_32 cpIndex)
    return *((uint64_t *) & romLiterals()[cpIndex]);
    }
 
+void *
+TR_ResolvedJ9Method::getConstantDynamicTypeFromCP(I_32 cpIndex)
+   {
+   return jitGetConstantDynamicTypeFromCP(fej9()->vmThread(), cp(), cpIndex);
+   }
+
+bool
+TR_ResolvedJ9Method::isConstantDynamic(I_32 cpIndex)
+   {
+   TR_ASSERT(cpIndex != -1, "cpIndex shouldn't be -1");
```

Comment:
This should probably be an TR_ASSERT_FATAL - it will never work with cpIndex -1, it doesn't handle the case, and it is just plain wrong to call it with -1. The check is cheap so keeping it in at runtime will make sure we don't miss a bad use of this API.

Reply:
Yes, this method should have fatal assertion as this query is the entry point of JIT condy processing logic.